lru_nots.go

```diff
@@ -97,6 +97,39 @@ func (l *LRUNoTS) Set(key string, val interface{}) error {
 	return nil
 }
 
+// SetNX sets a value to the cache only if it does not already exist
+func (l *LRUNoTS) SetNX(key string, val interface{}) (bool, error) {
+	// try to get item
+	_, err := l.cache.Get(key)
+	if err != nil && err != ErrNotFound {
+		return false, err
+	}
+
+	var elem *list.Element
+
+	// if elem is not in the cache, push it to front of the list
+	if err == ErrNotFound {
+		elem = l.list.PushFront(&kv{k: key, v: val})
+	} else {
+		// if elem is already in the cache, do nothing
+		return false, nil
+	}
+
```

Comment:
Done, and I actually reused Get too, because I think it's correct to move the element on the head of the list even if we are not setting it because already set.